lib/avalon/batch/entry.rb

```diff
@@ -18,126 +18,150 @@
-
-    	def initialize(fields, files, opts, row, manifest)
-	  @fields = fields
-	  @files  = files
-	  @opts   = opts
-	  @row    = row
-	  @manifest = manifest
-	  @errors = ActiveModel::Errors.new(self)
-	  @files.each { |file| file[:file] = File.join(@manifest.package.dir, file[:file]) }
-        end
-
-        def media_object
-          @media_object ||= MediaObject.new(avalon_uploader: @manifest.package.user.user_key, 
-                                            collection: @manifest.package.collection).tap do |mo|
-            mo.workflow.origin = 'batch'
-            if Avalon::BibRetriever.configured? and fields[:bibliographic_id].present?
-              begin
-                mo.descMetadata.populate_from_catalog!(fields[:bibliographic_id].first, Array(fields[:bibliographic_id_label]).first)
-              rescue Exception => e
-                @errors.add(:bibliographic_id, e.message)
-              end
-              # Sometimes we want to override the bib import data for some of the
-              # required fields (particularly if import data doesn't pass validations)
-              mo.update_datastream(:descMetadata,
-                                   fields.slice(:language,
-                                                :topical_subject,
-                                                :genre))
-            else
-              mo.update_datastream(:descMetadata, fields.dup)
+      extend ActiveModel::Translation
+
+      attr_reader :fields, :files, :opts, :row, :errors, :manifest, :collection
+
+      def initialize(fields, files, opts, row, manifest)
+        @fields = fields
+        @files  = files
+        @opts   = opts
+        @row    = row
+        @manifest = manifest
+        @errors = ActiveModel::Errors.new(self)
+        @files.each { |file| file[:file] = File.join(@manifest.package.dir, file[:file]) }
+      end
+
+      def media_object
+        return @media_object if @media_object
+        @media_object = nil
+        if fields[:media_object].present?
+          begin
+            @media_object = MediaObject.find(fields[:media_object].first)
+          rescue ActiveFedora::ObjectNotFoundError => e
+            @errors.add(:media_object, e.message)
+            return nil
+          end
+        else
+          @media_object = MediaObject.new(avalon_uploader: @manifest.package.user.user_key, 
+                                          collection: @manifest.package.collection)
+        end
+        update_media_object
+        return @media_object
+      end
+
+      def update_media_object
+        @media_object.tap do |mo|
+          mo.workflow.origin = 'batch'
+          if Avalon::BibRetriever.configured? and fields[:bibliographic_id].present?
+            begin
+              mo.descMetadata.
+                populate_from_catalog!(fields[:bibliographic_id].first,
+                                       Array(fields[:bibliographic_id_label]).first)
+            rescue Exception => e
```

Comment:
Should this be Exception, or StandardError? As written, this would result in, say, syntax errors in the begin block or the methods it calls being reported to the user as validation errors.